dll/win32/shell32/CCopyMoveToMenu.cpp

```diff
@@ -436,3 +436,145 @@ CCopyMoveToMenu::GetSite(REFIID riid, void **ppvSite)
 
     return m_pSite->QueryInterface(riid, ppvSite);
 }
+
+static DWORD
+SetClipboard(UINT cf, const void* data, SIZE_T size)
+{
+    BOOL succ = FALSE;
+    HGLOBAL handle = GlobalAlloc(GMEM_MOVEABLE | GMEM_SHARE, size);
+    if (handle)
+    {
+        LPVOID clipdata = GlobalLock(handle);
+        if (clipdata)
+        {
+            CopyMemory(clipdata, data, size);
+            GlobalUnlock(handle);
+            if (OpenClipboard(NULL))
+            {
+                EmptyClipboard();
+                succ = SetClipboardData(cf, handle) != NULL;
+                CloseClipboard();
+            }
+        }
+        if (!succ)
+        {
+            GlobalFree(handle);
+        }
+    }
+    return succ ? ERROR_SUCCESS : GetLastError();
+}
+
+static DWORD
+SetClipboardFromString(LPCWSTR str)
+{
+    SIZE_T cch = lstrlenW(str) + 1, size = cch * sizeof(WCHAR);
+    if (size > cch)
+        return SetClipboard(CF_UNICODETEXT, str, size);
+    else
+        return ERROR_BUFFER_OVERFLOW;
+}
+
+static void
+AppendToPathList(CStringW &paths, LPCWSTR path, DWORD index)
+{
+    if (index)
+        paths += L"\r\n";
+    LPCWSTR quote = StrChrW(path, L' ');
+    if (quote)
+        paths += L'\"';
+    paths += path;
+    if (quote)
+        paths += L'\"';
+}
+
+STDMETHODIMP
+CCopyAsPathMenu::Drop(IDataObject *pdto, DWORD grfKeyState, POINTL ptl, DWORD *pdwEffect)
+{
+    CStringW paths;
+    DWORD i, count;
+#if 0
```

Comment:
Can you wrap this in a Vista+ version check instead of just disabling it entirely like this.
The WIN32_VISTA_etc one will be good enough.

Reply:
`SHCreateShellItemArrayFromDataObject` does not actually exist yet in ROS Shell32 so that is why I disabled it outright. I can probably implement IShellItemArray later when other stuff starts using it.